WebSocket readLoop panics on timeout errors; Call() hangs when readLoop exits

## Description

Two bugs in `pkg/connection/gorillaws/connection.go` (v1.3.0) cause panics and permanent hangs when a WebSocket connection experiences network issues (timeouts, connection resets):

### Bug 1: `handleError()` does not recognize timeout errors

`handleError()` only checks for `net.ErrClosed` and `IsUnexpectedCloseError`. When the connection gets an `i/o timeout` (which is a `*net.OpError`), neither check matches, so `handleError` returns `false`. The `readLoop` then calls `conn.ReadMessage()` again on the dead connection, which eventually causes gorilla/websocket to panic internally.

**Current code:**

```go
func (c *Connection) handleError(err error) bool {
    if errors.Is(err, net.ErrClosed) {
        // ...
        return true
    }
    if gorilla.IsUnexpectedCloseError(err) {
        // ...
        return true
    }
    // timeout errors fall through here and return false
    c.logger.Error(err.Error())
    return false
}
```

**Expected:** `*net.OpError` and `os.ErrDeadlineExceeded` should be treated as fatal, causing `readLoop` to exit cleanly via `closeWithError()`.

### Bug 2: `Call()` hangs forever when `readLoop` exits

The `Call()` method waits for a response on a channel that is only written to by `readLoop`. When `readLoop` exits (e.g. due to Bug 1 or a legitimate close), in-flight `Call()` requests block forever because `connCloseCh` is not checked in the final select:

**Current code:**

```go
select {
case <-ctx.Done():
    return nil, ctx.Err()
case res, open := <-responseChan:
    // ...
}
```

**Expected:** The select should also check `connCloseCh`:

```go
select {
case <-ctx.Done():
    return nil, ctx.Err()
case <-c.connCloseCh:
    return nil, c.connCloseError
case res, open := <-responseChan:
    // ...
}
```

Without this, any goroutine that has a `Call()` in flight when the connection drops will hang permanently.

## Impact

- **Panic in `readLoop`**: Crashes the entire Go process. Since `readLoop` runs in a goroutine spawned by `Connect()`, the caller cannot `recover()` from it.
- **Hang in `Call()`**: Any in-flight RPC call blocks forever, causing goroutine leaks and eventual resource exhaustion.

These are particularly impactful for long-lived connections (e.g. API servers holding persistent connections to SurrealDB Cloud), where transient network issues are expected.

## Steps to Reproduce

1. Establish a WebSocket connection to SurrealDB Cloud
2. Start a long-running operation that sends many queries (e.g. bulk inserts)
3. Wait for a network interruption (i/o timeout, connection reset)
4. Observe: `readLoop` panics inside gorilla/websocket, or `Call()` hangs permanently

## Environment

- SDK version: v1.3.0
- SurrealDB: Cloud (v3.0.0)
- Go: 1.24
- OS: Linux (also reproducible on macOS)

## Suggested Fix

1. In `handleError()`, add checks for `*net.OpError` and `os.ErrDeadlineExceeded` that return `true` (and call `closeWithError`)
2. In `Call()`, add `case <-c.connCloseCh` to the response-wait select to unblock when the connection dies

Experiencing exactly this issue. It effectively crashes the app and makes unreliable for any kind of production environment.

{"time":"2026-04-14T08:09:25.097942995+02:00","level":"ERROR","msg":"read tcp 172.17.0.249:54648->51.44.42.61:443: read: connection timed out"}
{"time":"2026-04-14T08:09:25.097946752+02:00","level":"ERROR","msg":"read tcp 172.17.0.249:54648->51.44.42.61:443: read: connection timed out"}
panic: repeated read on failed websocket connection

goroutine 26 [running]:
github.com/gorilla/websocket.(*Conn).NextReader(0xc00016c2c0)
	/home/iloginow/go/pkg/mod/github.com/gorilla/websocket@v1.5.3/conn.go:1030 +0x1fb
github.com/gorilla/websocket.(*Conn).ReadMessage(0xc000176200?)
	/home/iloginow/go/pkg/mod/github.com/gorilla/websocket@v1.5.3/conn.go:1093 +0x13
github.com/surrealdb/surrealdb.go/pkg/connection/gorillaws.(*Connection).readLoop(0xc000176200, 0xc00016c2c0)
	/home/iloginow/go/pkg/mod/github.com/surrealdb/surrealdb.go@v1.4.0/pkg/connection/gorillaws/connection.go:434 +0x45
created by github.com/surrealdb/surrealdb.go/pkg/connection/gorillaws.(*Connection).Connect in goroutine 1
	/home/iloginow/go/pkg/mod/github.com/surrealdb/surrealdb.go@v1.4.0/pkg/connection/gorillaws/connection.go:138 +0x2de
Process Exit with Code: 2

--- a/pkg/connection/gorillaws/connection.go
+++ b/pkg/connection/gorillaws/connection.go
@@ -462,6 +462,16 @@ func (c *Connection) handleError(err error) bool {
 		return true
 	}
 
+	// Handle timeout and network errors to prevent readLoop from
+	// retrying on a dead connection, which can cause panics.
+	var netErr net.Error
+	if errors.As(err, &netErr) {
+		c.closeMutex.Lock()
+		c.connCloseError = err
+		c.closeMutex.Unlock()
+		return true
+	}
+
 	c.logger.Error(err.Error())
 	return false
 }

